source/Calamari.Tests/KubernetesFixtures/Helm/HelmTemplateValueSourcesCreatorFixture.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Calamari.Common.Commands;
+using Calamari.Common.Plumbing.Deployment;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Kubernetes;
+using Calamari.Kubernetes.Helm;
+using Calamari.Tests.Helpers;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Newtonsoft.Json;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Calamari.Tests.KubernetesFixtures.Helm
+{
+    [TestFixture]
+    public class HelmTemplateValueSourcesCreatorFixture
+    {
+        static readonly string RootDir = Path.Combine("root", "staging");
+
+        [Test]
+        public void ParseTemplateValuesSources_ChartSourceButIncorrectScriptSource_ShouldThrowArgumentException()
+        {
+            // Arrange
+            var templateValuesSourcesJson = JsonConvert.SerializeObject(new HelmTemplateValueSourcesParser.TemplateValuesSource[]
+                                                                        {
+                                                                            new HelmTemplateValueSourcesParser.ChartTemplateValuesSource
+                                                                            {
+                                                                                ValuesFilePaths = "secondary.yaml"
+                                                                            }
+                                                                        },
+                                                                        Formatting.None);
+
+            var variables = new CalamariVariables
+            {
+                [SpecialVariables.Helm.TemplateValuesSources] = templateValuesSourcesJson,
+                [KnownVariables.OriginalPackageDirectoryPath] = RootDir,
+                [ScriptVariables.ScriptSource] = ScriptVariables.ScriptSourceOptions.Core,
+                [PackageVariables.IndexedPackageId(string.Empty)] = "mychart",
+                [PackageVariables.IndexedPackageVersion(string.Empty)] = "0.3.8"
+            };
+
+            var deployment = new RunningDeployment(variables)
+            {
+                CurrentDirectoryProvider = DeploymentWorkingDirectory.StagingDirectory
+            };
+            var fileSystem = Substitute.For<ICalamariFileSystem>();
+
+            // Act
+            Action act = () => HelmTemplateValueSourcesParser.ParseTemplateValuesSources(deployment, fileSystem, new SilentLog());
+
+            // Assert
+            act.Should().ThrowExactly<ArgumentException>();
+        }
+        
+        [Test]
+        public void ParseTemplateValuesSources_ChartSourceButNotScriptSourceVairable_ShouldThrowArgumentNullException()
```

Comment:
`Vairable`

Reply:
I'll fix in a future PR, just want to merge this one